### PR TITLE
feat(keycloak): add a new database

### PR DESCRIPTION
**Pull request by smerle33**
create a dedicated bdd on the flexible postgresql public instance on azure.

Ref: https://github.com/jenkins-infra/helpdesk/issues/3214

### CODE REVIEW

**Comment by dduportal** [comment]
@smerle33 this PR has conflicts. Can you rebase it?

File: keycloak.jenkins.io.tf [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+output "keycloak_dbconfig" {
+  sensitive   = true
+  description = "YAML (secret) values for the Helm chart codecentric/keycloak"
+  value       = <<-EOT
+secrets:
+    db:
+        data:
+            DB_USER: ${base64encode(postgresql_role.keycloak.name)}
+            DB_PASSWORD: ${base64encode(random_password.pgsql_keycloak_user_password.result)}
+            DB_VENDOR: cG9zdGdyZXM=
```
Comment (lemeurherve) [inline keycloak.jenkins.io.tf:28]:
```suggestion
            DB_VENDOR: ${base64encode("postgres")}
```

File: keycloak.jenkins.io.tf [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  special          = true
+}
+
+resource "postgresql_role" "keycloak" {
+  name     = "keycloak"
+  login    = true
+  password = random_password.pgsql_keycloak_user_password.result
+}
+
+# This (sensitive) output is meant to be encrypted into the production secret system, to be provided as a secret to the keycloaks application https://admin.accounts.jenkins.io/auth/admin
```
Comment (lemeurherve) [inline keycloak.jenkins.io.tf:18]:
```suggestion
# This (sensitive) output is meant to be encrypted into the production secret system, to be provided as a secret to the Keycloak application (https://admin.accounts.jenkins.io)
```

File: keycloak.jenkins.io.tf [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+  description = "YAML (secret) values for the Helm chart codecentric/keycloak"
+  value       = <<-EOT
+secrets:
+    db:
+        data:
+            DB_USER: ${base64encode(postgresql_role.keycloak.name)}
+            DB_PASSWORD: ${base64encode(random_password.pgsql_keycloak_user_password.result)}
+            DB_VENDOR: cG9zdGdyZXM=
+            DB_ADDR: ${base64encode(azurerm_postgresql_flexible_server.public.fqdn)}
+            DB_PORT: NTQzMg==
```
Comment (lemeurherve) [inline keycloak.jenkins.io.tf:30]:
```suggestion
            DB_PORT: ${base64encode("5432")}
```

File: keycloak.jenkins.io.tf [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+  length           = 24
+  override_special = "!#%&*()-_=+[]{}:?"
+  special          = true
+}
+
+resource "postgresql_role" "keycloak" {
+  name     = "keycloak"
+  login    = true
+  password = random_password.pgsql_keycloak_user_password.result
+}
```
Comment (lemeurherve) [inline keycloak.jenkins.io.tf:16]:
```suggestion
resource "random_password" "pgsql_keycloak_user_password" {
  length           = 24
  override_special = "!#%&*()-_=+[]{}:?"
  special          = true
}

resource "postgresql_role" "keycloak" {
  name     = "keycloak"
  login    = true
  password = random_password.pgsql_keycloak_user_password.result
}

resource "postgresql_database" "keycloak" {
  name  = "keycloak"
  owner = postgresql_role.keycloak.name
}
```
(nit) easier with the declaration of element first then their use in another element.

**Review by lemeurherve — DISMISSED** [review]

File: keycloak.jenkins.io.tf [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+
+# This (sensitive) output is meant to be encrypted into the production secret system, to be provided as a secret to the Keycloak application (https://admin.accounts.jenkins.io)
+output "keycloak_dbconfig" {
+  # default port to use: https://learn.microsoft.com/en-us/azure/postgresql/flexible-server/concepts-networking
+  # nothing to get from terraform for the port: https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/postgresql_flexible_server#attributes-reference
```
Comment (dduportal) [inline keycloak.jenkins.io.tf:21]:
```suggestion
  # Value of DB_PORT: 5432 is the only usable port: https://learn.microsoft.com/en-us/azure/postgresql/flexible-server/concepts-networking
  ## Terraform resource does not export any port attribute: https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/postgresql_flexible_server#attributes-reference
```

**Review by dduportal — APPROVED** [review]
`Plan: 3 to add, 0 to change, 0 to destroy.`

LGTM

**Pull request merged**